### ISSUE TITLE
ssr doesn't recognize lib shortnames in tsconfig

### ISSUE THREAD

**Issue by goelinsights** [nrwl/nx#574]
Have an angular universal app where bundling to prod works fine for import statements with @lib, but getting the error
`Could not resolve @workpace/mylib-name relative to /Users/me/folder/workspace/apps/app-name/src/app/app.module.ts`

Is there a way to do this, or do I need to turn library @names into relative paths?

Seems like the server side rendering isn't able to pull the paths from tsconfig

**Comment by FrozenPandaz** [comment]
What a coincidence, just ran into this today. 

It's an issue with the schematic that gets generated. Make sure you remove the baseUrl property in the `tsconfig.server.json` and extend either off of the `tsconfig.app.json` or the root `tsconfig.json`.

So the workaround is manually reconfiguring it. Let's also leave this issue open to track resolution of the issue.

**Comment by goelinsights** [comment]
That did it!

**Comment by hmagrini** [comment]
I stumbled on this ticket cause I had a similar issue when trying to pre-render my application and compiling the server with Webpack as per the CLI's [Universal story](https://github.com/angular/angular-cli/wiki/stories-universal-rendering#step-5-setup-a-webpack-config-to-handle-this-node-serverts-file-and-serve-your-application) due to Webpack not being able to resolve the `paths` config from the `tsconfig.json` file.
Since this ticket it's related to SSR and other people might also stumble on this, I thought I might share that, to solve this, you can use `TsconfigPathsPlugin` from [tsconfig-paths-webpack-plugin](https://github.com/dividab/tsconfig-paths-webpack-plugin) and use it on your `webpack.config.js` like this:

```
resolve: {
  extensions: ['.js', '.ts'],
  plugins: [
    new TsconfigPathsPlugin({
      configFile: 'path/to/tsconfig.server.json`)
    })
  ]
},
```

**Comment by miguelchico** [comment]
@hmagrini 
Thank you very much for it. Using the TsconfigPathsPlugin I was able to fix my issue.

**Comment by vggarg96** [comment]
@FrozenPandaz @hmagrini i am facing the similar problem. From the above solution i was able to create the build but when i was running the build i got the following error:



`$ npm run serve:ssr

> myworkspace@0.0.0 serve:ssr E:\vaibhav\bag2bag\myworkspace
> node dist/apps/server

Node Express server listening on http://localhost:4000
ERROR { Error: Uncaught (in promise): Error: Cannot find module '@myworkspace/mylib/src/lib/mylib.module.ngfactory'
Error: Cannot find module '@myworkspace/mylib/src/lib/mylib.module.ngfactory'
    at E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:39145:11
    at ZoneDelegate.invoke (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:536:26)
    at Object.onInvoke (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:20256:33)
    at ZoneDelegate.invoke (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:535:52)
    at Zone.run (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:295:43)
    at E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:1034:34
    at ZoneDelegate.invokeTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:568:31)
    at Object.onInvokeTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:20247:33)
    at ZoneDelegate.invokeTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:567:60)
    at Zone.runTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:340:47)
    at resolvePromise (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:976:31)
    at resolvePromise (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:933:17)
    at E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:1037:17
    at ZoneDelegate.invokeTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:568:31)
    at Object.onInvokeTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:20247:33)
    at ZoneDelegate.invokeTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:567:60)
    at Zone.runTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:340:47)
    at drainMicroTaskQueue (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:746:35)
    at ZoneTask.invokeTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:647:21)
    at Server.ZoneTask.invoke (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:632:48)
  rejection:
   { Error: Cannot find module '@myworkspace/mylib/src/lib/mylib.module.ngfactory'
       at E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:39145:11
       at ZoneDelegate.invoke (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:536:26)
       at Object.onInvoke (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:20256:33)
       at ZoneDelegate.invoke (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:535:52)
       at Zone.run (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:295:43)
       at E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:1034:34
       at ZoneDelegate.invokeTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:568:31)
       at Object.onInvokeTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:20247:33)
       at ZoneDelegate.invokeTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:567:60)
       at Zone.runTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:340:47) code: 'MODULE_NOT_FOUND' },
  promise:
   ZoneAwarePromise {
     __zone_symbol__state: 0,
     __zone_symbol__value:
      { Error: Cannot find module '@myworkspace/mylib/src/lib/mylib.module.ngfactory'
          at E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:39145:11
          at ZoneDelegate.invoke (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:536:26)
          at Object.onInvoke (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:20256:33)
          at ZoneDelegate.invoke (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:535:52)
          at Zone.run (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:295:43)
          at E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:1034:34
          at ZoneDelegate.invokeTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:568:31)
          at Object.onInvokeTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:20247:33)
          at ZoneDelegate.invokeTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:567:60)
          at Zone.runTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:340:47) code: 'MODULE_NOT_FOUND' } },
  zone:
   Zone {
     _parent:
      Zone {
        _parent: null,
        _name: '<root>',
        _properties: {},
        _zoneDelegate: [ZoneDelegate] },
     _name: 'angular',
     _properties: { isAngularZone: true },
     _zoneDelegate:
      ZoneDelegate {
        _taskCounts: [Object],
        zone: [Circular],
        _parentDelegate: [ZoneDelegate],
        _forkZS: null,
        _forkDlgt: null,
        _forkCurrZone: [Zone],
        _interceptZS: null,
        _interceptDlgt: null,
        _interceptCurrZone: [Zone],
        _invokeZS: [Object],
        _invokeDlgt: [ZoneDelegate],
        _invokeCurrZone: [Circular],
        _handleErrorZS: [Object],
        _handleErrorDlgt: [ZoneDelegate],
        _handleErrorCurrZone: [Circular],
        _scheduleTaskZS: [Object],
        _scheduleTaskDlgt: [ZoneDelegate],
        _scheduleTaskCurrZone: [Circular],
        _invokeTaskZS: [Object],
        _invokeTaskDlgt: [ZoneDelegate],
        _invokeTaskCurrZone: [Circular],
        _cancelTaskZS: [Object],
        _cancelTaskDlgt: [ZoneDelegate],
        _cancelTaskCurrZone: [Circular],
        _hasTaskZS: [Object],
        _hasTaskDlgt: [ZoneDelegate],
        _hasTaskDlgtOwner: [Circular],
        _hasTaskCurrZone: [Circular] } },
  task:
   ZoneTask {
     _zone:
      Zone {
        _parent: [Zone],
        _name: 'angular',
        _properties: [Object],
        _zoneDelegate: [ZoneDelegate] },
     runCount: 0,
     _zoneDelegates: null,
     _state: 'notScheduled',
     type: 'microTask',
     source: 'Promise.then',
     data:
      ZoneAwarePromise { __zone_symbol__state: 0, __zone_symbol__value: [Error] },
     scheduleFn: undefined,
     cancelFn: undefined,
     callback: [Function],
     invoke: [Function] } }
ERROR { Error: Uncaught (in promise): Error: Cannot find module '@myworkspace/mylib/src/lib/mylib.module.ngfactory'
Error: Cannot find module '@myworkspace/mylib/src/lib/mylib.module.ngfactory'
    at E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:39145:11
    at ZoneDelegate.invoke (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:536:26)
    at Object.onInvoke (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:20256:33)
    at ZoneDelegate.invoke (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:535:52)
    at Zone.run (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:295:43)
    at E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:1034:34
    at ZoneDelegate.invokeTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:568:31)
    at Object.onInvokeTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:20247:33)
    at ZoneDelegate.invokeTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:567:60)
    at Zone.runTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:340:47)
    at resolvePromise (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:976:31)
    at resolvePromise (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:933:17)
    at E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:1037:17
    at ZoneDelegate.invokeTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:568:31)
    at Object.onInvokeTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:20247:33)
    at ZoneDelegate.invokeTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:567:60)
    at Zone.runTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:340:47)
    at drainMicroTaskQueue (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:746:35)
    at ZoneTask.invokeTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:647:21)
    at Server.ZoneTask.invoke (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:632:48)
  rejection:
   { Error: Cannot find module '@myworkspace/mylib/src/lib/mylib.module.ngfactory'
       at E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:39145:11
       at ZoneDelegate.invoke (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:536:26)
       at Object.onInvoke (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:20256:33)
       at ZoneDelegate.invoke (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:535:52)
       at Zone.run (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:295:43)
       at E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:1034:34
       at ZoneDelegate.invokeTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:568:31)
       at Object.onInvokeTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:20247:33)
       at ZoneDelegate.invokeTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:567:60)
       at Zone.runTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:340:47) code: 'MODULE_NOT_FOUND' },
  promise:
   ZoneAwarePromise {
     __zone_symbol__state: 0,
     __zone_symbol__value:
      { Error: Cannot find module '@myworkspace/mylib/src/lib/mylib.module.ngfactory'
          at E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:39145:11
          at ZoneDelegate.invoke (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:536:26)
          at Object.onInvoke (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:20256:33)
          at ZoneDelegate.invoke (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:535:52)
          at Zone.run (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:295:43)
          at E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:1034:34
          at ZoneDelegate.invokeTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:568:31)
          at Object.onInvokeTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:20247:33)
          at ZoneDelegate.invokeTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:567:60)
          at Zone.runTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:340:47) code: 'MODULE_NOT_FOUND' } },
  zone:
   Zone {
     _parent:
      Zone {
        _parent: null,
        _name: '<root>',
        _properties: {},
        _zoneDelegate: [ZoneDelegate] },
     _name: 'angular',
     _properties: { isAngularZone: true },
     _zoneDelegate:
      ZoneDelegate {
        _taskCounts: [Object],
        zone: [Circular],
        _parentDelegate: [ZoneDelegate],
        _forkZS: null,
        _forkDlgt: null,
        _forkCurrZone: [Zone],
        _interceptZS: null,
        _interceptDlgt: null,
        _interceptCurrZone: [Zone],
        _invokeZS: [Object],
        _invokeDlgt: [ZoneDelegate],
        _invokeCurrZone: [Circular],
        _handleErrorZS: [Object],
        _handleErrorDlgt: [ZoneDelegate],
        _handleErrorCurrZone: [Circular],
        _scheduleTaskZS: [Object],
        _scheduleTaskDlgt: [ZoneDelegate],
        _scheduleTaskCurrZone: [Circular],
        _invokeTaskZS: [Object],
        _invokeTaskDlgt: [ZoneDelegate],
        _invokeTaskCurrZone: [Circular],
        _cancelTaskZS: [Object],
        _cancelTaskDlgt: [ZoneDelegate],
        _cancelTaskCurrZone: [Circular],
        _hasTaskZS: [Object],
        _hasTaskDlgt: [ZoneDelegate],
        _hasTaskDlgtOwner: [Circular],
        _hasTaskCurrZone: [Circular] } },
  task:
   ZoneTask {
     _zone:
      Zone {
        _parent: [Zone],
        _name: 'angular',
        _properties: [Object],
        _zoneDelegate: [ZoneDelegate] },
     runCount: 0,
     _zoneDelegates: null,
     _state: 'notScheduled',
     type: 'microTask',
     source: 'Promise.then',
     data:
      ZoneAwarePromise { __zone_symbol__state: 0, __zone_symbol__value: [Error] },
     scheduleFn: undefined,
     cancelFn: undefined,
     callback: [Function],
     invoke: [Function] } }
ERROR { Error: Uncaught (in promise): Error: Cannot find module '@myworkspace/mylib/src/lib/mylib.module.ngfactory'
Error: Cannot find module '@myworkspace/mylib/src/lib/mylib.module.ngfactory'
    at E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:39145:11
    at ZoneDelegate.invoke (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:536:26)
    at Object.onInvoke (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:20256:33)
    at ZoneDelegate.invoke (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:535:52)
    at Zone.run (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:295:43)
    at E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:1034:34
    at ZoneDelegate.invokeTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:568:31)
    at Object.onInvokeTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:20247:33)
    at ZoneDelegate.invokeTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:567:60)
    at Zone.runTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:340:47)
    at resolvePromise (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:976:31)
    at resolvePromise (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:933:17)
    at E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:1037:17
    at ZoneDelegate.invokeTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:568:31)
    at Object.onInvokeTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:20247:33)
    at ZoneDelegate.invokeTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:567:60)
    at Zone.runTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:340:47)
    at drainMicroTaskQueue (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:746:35)
    at ZoneTask.invokeTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:647:21)
    at Server.ZoneTask.invoke (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:632:48)
  rejection:
   { Error: Cannot find module '@myworkspace/mylib/src/lib/mylib.module.ngfactory'
       at E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:39145:11
       at ZoneDelegate.invoke (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:536:26)
       at Object.onInvoke (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:20256:33)
       at ZoneDelegate.invoke (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:535:52)
       at Zone.run (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:295:43)
       at E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:1034:34
       at ZoneDelegate.invokeTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:568:31)
       at Object.onInvokeTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:20247:33)
       at ZoneDelegate.invokeTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:567:60)
       at Zone.runTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:340:47) code: 'MODULE_NOT_FOUND' },
  promise:
   ZoneAwarePromise {
     __zone_symbol__state: 0,
     __zone_symbol__value:
      { Error: Cannot find module '@myworkspace/mylib/src/lib/mylib.module.ngfactory'
          at E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:39145:11
          at ZoneDelegate.invoke (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:536:26)
          at Object.onInvoke (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:20256:33)
          at ZoneDelegate.invoke (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:535:52)
          at Zone.run (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:295:43)
          at E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:1034:34
          at ZoneDelegate.invokeTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:568:31)
          at Object.onInvokeTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:20247:33)
          at ZoneDelegate.invokeTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:567:60)
          at Zone.runTask (E:\vaibhav\bag2bag\myworkspace\dist\apps\server.js:340:47) code: 'MODULE_NOT_FOUND' } },
  zone:
   Zone {
     _parent:
      Zone {
        _parent: null,
        _name: '<root>',
        _properties: {},
        _zoneDelegate: [ZoneDelegate] },
     _name: 'angular',
     _properties: { isAngularZone: true },
     _zoneDelegate:
      ZoneDelegate {
        _taskCounts: [Object],
        zone: [Circular],
        _parentDelegate: [ZoneDelegate],
        _forkZS: null,
        _forkDlgt: null,
        _forkCurrZone: [Zone],
        _interceptZS: null,
        _interceptDlgt: null,
        _interceptCurrZone: [Zone],
        _invokeZS: [Object],
        _invokeDlgt: [ZoneDelegate],
        _invokeCurrZone: [Circular],
        _handleErrorZS: [Object],
        _handleErrorDlgt: [ZoneDelegate],
        _handleErrorCurrZone: [Circular],
        _scheduleTaskZS: [Object],
        _scheduleTaskDlgt: [ZoneDelegate],
        _scheduleTaskCurrZone: [Circular],
        _invokeTaskZS: [Object],
        _invokeTaskDlgt: [ZoneDelegate],
        _invokeTaskCurrZone: [Circular],
        _cancelTaskZS: [Object],
        _cancelTaskDlgt: [ZoneDelegate],
        _cancelTaskCurrZone: [Circular],
        _hasTaskZS: [Object],
        _hasTaskDlgt: [ZoneDelegate],
        _hasTaskDlgtOwner: [Circular],
        _hasTaskCurrZone: [Circular] } },
  task:
   ZoneTask {
     _zone:
      Zone {
        _parent: [Zone],
        _name: 'angular',
        _properties: [Object],
        _zoneDelegate: [ZoneDelegate] },
     runCount: 0,
     _zoneDelegates: null,
     _state: 'notScheduled',
     type: 'microTask',
     source: 'Promise.then',
     data:
      ZoneAwarePromise { __zone_symbol__state: 0, __zone_symbol__value: [Error] },
     scheduleFn: undefined,
     cancelFn: undefined,
     callback: [Function],
     invoke: [Function] } }

`


here is the link to my repo [https://github.com/vggarg96/nrwl-angular-universal](url)

**Comment by hmagrini** [comment]
@vggarg96 have you pushed all your changes? cause I see you have installed `TsconfigPathsPlugin` but you haven't added it to the `webpack.server.config.js` file as my comment suggests.

**Comment by vggarg96** [comment]
@hmagrini I am not getting this error while building. I am getting this error while running the build. I have removed `TsconfigPathsPlugin` code from the `webpack.server.config.js` because I am getting same error while running the build.

**Comment by hmagrini** [comment]
> I am getting this error while running the build

@vggarg96 hmm maybe you can try installing [`tsconfig-paths`](https://www.npmjs.com/package/tsconfig-paths) and change your `serve:ssr` npm script to 
```diff
+ node -r tsconfig-paths/register dist/apps/server`
- node dist/apps/server
```

**Comment by vggarg96** [comment]
@hmagrini  still the same result i was able to build it but when i run it. i got the same error as before

**Comment by hmagrini** [comment]
@vggarg96 I opened a PR to your repo (https://github.com/vggarg96/nrwl-angular-universal/pull/1). I think the most important part was that you were missing to use the `ModuleMapLoaderModule` like on the `app.server.module` file, like this: https://github.com/hmagrini/nrwl-angular-universal/blob/hmagrini/fix-build/apps/myworkspace/src/app/app.server.module.ts#L3

But please check the whole PR as I added a couple of things to make the setup more similar to mine (since mine was working).

**Comment by vggarg96** [comment]
thanks @hmagrini I am looking at the PR. Thanks a lot for your help and support. I will try to run it and let you know ASAP about the result  . Sorry for troubling I am still a college student and learning angular. Once again thank you a lot for helping me out.

**Comment by hmagrini** [comment]
no worries, happy to help! I'll add some comments to the PR to explain a bit some of the changes :)

**Comment by vsavkin** [comment]
@FrozenPandaz can you take a look?

**Comment by NateVonSmith** [comment]
For anybody still struggling with this (thanks for your suggestions @hmagrini but I also couldn't get it to work - same situation as @vggarg96), I ended up just adding a build step at the end to replace `href="styles.` in the built index.html file with `href="/styles.` (root-relative) to resolve the issues I was having where css file paths that were relative to the url routes, which I assume is the result of not including the baseUrl param in webpack.server.config.

Tacked this onto my npm build script ` && node scripts/adjust-ng-build` (file below is located at `scripts/adjust-ng-build.js` in my project) which runs..

```const replace = require('replace-in-file');
const options = {
    files: 'dist/apps/PROJECT_NAME/index.html',
    from: 'href="styles.',
    to: 'href="/styles.'
};

replace(options)
    .then(results => {
        console.log('Replacement results:', results);
    })
    .catch(error => {
        console.error('Error occurred:', error);
    });
```

Def a hacky solution but got me over the hump so figured I'd share for anyone else who's blocked by it.
Take care!

**Comment by FrozenPandaz** [comment]
v8 of Angular no longer generates the `baseUrl` compiler option. 🎉Thus, I'm going to close this issue.

As per the paths.. It seems to be working for me.. Can someone open a new issue please and provide a repo?
